Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Operator/Driver.hs

```diff
@@ -98,10 +102,21 @@ postDriverOperatorRespondHubRequest merchantShortId opCity req = do
   Redis.whenWithLockRedis (opsHubRequestLockKey req.operationHubRequestId) 60 $ do
     opHubReq <- SQOHR.findByPrimaryKey (Kernel.Types.Id.Id req.operationHubRequestId) >>= fromMaybeM (InternalError "Invalid operation hub request id")
     unless (opHubReq.requestStatus == PENDING) $ Kernel.Utils.Common.throwError (InvalidRequest "Request already responded")
-    void $ SQOHR.updateStatusWithDetails (castReqStatusToDomain req.status) (Just req.remarks) (Just now) (Just (Kernel.Types.Id.Id req.operatorId)) (Kernel.Types.Id.Id req.operationHubRequestId)
     when (req.status == API.Types.ProviderPlatform.Operator.Driver.APPROVED && opHubReq.requestType == ONBOARDING_INSPECTION) $ do
       fork "enable driver after inspection" $ do
-        let personId = opHubReq.driverId
+        creator <- runInReplica $ QPerson.findById opHubReq.creatorId >>= fromMaybeM (PersonNotFound opHubReq.creatorId.getId)
+        personId <- case creator.role of
+          DP.DRIVER -> pure creator.id
+          DP.OPERATOR -> do
+            rc <- QVRCE.findLastVehicleRCWrapper opHubReq.registrationNo >>= fromMaybeM (RCNotFound opHubReq.registrationNo)
+            drc <- SQDRA.findAllActiveAssociationByRCId rc.id
+            case drc of
+              [] -> throwError (InvalidRequest "No driver exist with this RC")
+              (assoc : _) -> do
+                isAssociated <- DDriver.checkDriverOperatorAssociation assoc.driverId opHubReq.creatorId
```

Comment:
should we consider here indirect association **driver <-> fleet + fleet <-> operator**? But anyway it can be implemented separately